corehq/messaging/scheduling/views.py

```diff
@@ -1085,3 +1092,48 @@ def post(self, request, *args, **kwargs):
             msg[0](request, msg[1])
 
         return self.get(request, *args, **kwargs)
+
+
+@login_and_domain_required
+def messaging_image_upload_view(request, domain):
+    if request.method == 'POST' and request.FILES.get('upload'):
+        image_file = request.FILES['upload']
+
+        if image_file.size > MAX_IMAGE_UPLOAD_SIZE:
+            return JsonResponse({
+                'error': {
+                    "message": _('Image file is too large. Images must be smaller than 1MB'),
+                }
+            }, status=400)
+
+        if image_file.content_type not in VALID_EMAIL_IMAGE_MIMETYPES:
+            image_extensions = [mimetype.split("/")[1] for mimetype in VALID_EMAIL_IMAGE_MIMETYPES]
+            return JsonResponse({
+                'error': {
+                    "message": _('You can only upload {image_extensions} images').format(
+                        image_extensions=", ".join(image_extensions))
+                },
+            }, status=400)
+
+        image = EmailImage.save_blob(
+            image_file,
+            domain=request.domain,
+            filename=image_file.name,
+            content_type=image_file.content_type,
+            delete_after=datetime.utcnow() + timedelta(days=365),
```

Comment:
Is this expiration date communicated to the user?

Reply:
This is something I wanted to talk about. In the spec, I suggested that there was no expiration date on these images, which I still think could be the right idea here. I ended up adding this expiration out of caution, but now that I think about it, it could be totally possible to define a conditional alert that runs for longer than a year.
I will remove this condition - does that sound right to you?

Reply:
https://github.com/dimagi/commcare-hq/pull/33798/commits/74876532ff71ef98d2101367428fa1e7ef73363a

Reply:
yeah, I agree with that - seems safer, especially if there's a cleanup process